cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -296,6 +303,49 @@ where
 			.ok_or(sp_blockchain::Error::Backend(format!("Header not found for {:?}", at)))
 	}
 
+	fn block_body(&self, at: Block::Hash) -> Result<Vec<Block::Extrinsic>, sp_blockchain::Error> {
+		self.client
+			.block_body(&BlockId::Hash(at))?
+			.ok_or(sp_blockchain::Error::Backend(format!("Block body not found for {:?}", at)))
+	}
+
+	fn create_extrinsic_execution_proof(
+		&self,
+		extrinsic_index: usize,
+		parent_header: &Block::Header,
+		current_hash: Block::Hash,
+	) -> Result<StorageProof, GossipMessageError> {
+		let extrinsics = self.block_body(current_hash)?;
+
+		let encoded_extrinsic = extrinsics[extrinsic_index].encode();
```

Comment:
These offset indexes can panic. Should we use `get` instead or, better yet, receive extrinsic itself as an argument so we don't have to retrieve it from storage in case we already have it somewhere?